### PR TITLE
added tests for ShareTensors

### LINKED ISSUE #23: Add tests for ShareTensors
## Description
Add tests for ShareTensors

## Type of Test
- [ ] **Unit test** (e.g. checking a loop, method, or function is working as intended)
- [ ] **Integration test** (e.g. checking if a certain group or set of functionality is working as intended)
- [ ] **Regression test** (e.g. checking if by adding or removing a module of code allows other systems to continue to function as intended)
- [ ] **Stress test** (e.g. checking to see how well a system performs under various situations, including heavy usage)
- [ ] **Performance test** (e.g. checking to see how efficient a system is as performing the intended task)
- [ ] Other...

## Expected Behavior
A clear and concise description of what you expected to happen. Do you intend to reach a certain percentage of test coverage for this file, feature, or codebase?

## Additional Context
Add any other context about the tests here.

### PR DESCRIPTION
**Pull request by aanurraj**
## Description
Added tests for ShareTensors
fixes #23 
## Affected Dependencies
None

## How has this been tested?
added tests

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tests/sympc/tensor/share_tensor_test.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
+import pytest
+import syft as sy
+import torch
+
+from sympc.session import Session
+from sympc.tensor import ShareTensor
+
+
+@pytest.fixture
+def clients():
```
Comment (LaRiffle) [inline tests/sympc/tensor/share_tensor_test.py:10]: Maybe this fixture should be in tests/conftest.py like in https://github.com/OpenMined/PySyft/blob/syft_0.2.x/test/conftest.py

Comment (aanurraj) [reply tests/sympc/tensor/share_tensor_test.py:10]: Yes will do it!

**Comment by gmuraru** [comment]
@aanurraj @LaRiffle @arturomf94 this is ready for review :D

**Comment by arturomf94** [comment]
It looks like `isort` is failing.

**Comment by arturomf94** [comment]
It looks great! Just an observation: It seems like there are some missing statements here on lines 173-176:

```
src/sympc/tensor/share_tensor.py           95      0     18      1    99%   173->176
```

So... `rmatmul`?

**Comment by gmuraru** [comment]
> It looks great! Just an observation: It seems like there are some missing statements here on lines 173-176:
> 
> ```
> src/sympc/tensor/share_tensor.py           95      0     18      1    99%   173->176
> ```
> 
> So... `rmatmul`?

I think it is because it always goes a specific path - fixing it now :D

**Review by arturomf94 — APPROVED** [review]
LGTM! :+1: :smiley:

**Comment by aanurraj** [comment]
Looks Good! 👍🏻
